src/Stmt.cc

```diff
@@ -2071,4 +2071,13 @@ TraversalCode WhenStmt::Traverse(TraversalCallback* cb) const {
     HANDLE_TC_STMT_POST(tc);
 }
 
+ValPtr StdFunctionStmt::Exec(Frame* f, StmtFlowType& flow) {
+    flow = FLOW_NEXT;
+    zeek::Args args = *f->GetFuncArgs();
+
+    func(args);
```

Comment:
I do wonder if we should support the `StmtFlowType& flow` in the std::function as well. Maybe with some sort of overload/variant so that you would not need to have it. Not sure if that's worth it.

Then you could actually use StdFunctionStmt with hook handlers and actually "break" from them.

Reply:
I can't override it because the function gets stored, so StdFunctionStmt would have to store both versions. I could see the function definition just always taking the flow though. Implementers would need to know to set it to `FLOW_NEXT` if they don't want to handle it though.

Reply:
I was thinking something like the following, allowing users to provide either variant and then calling the right one at Exec() time. Untested and not so sure about the names, but think this should work.

```
diff --git a/src/Stmt.cc b/src/Stmt.cc
index a75d768a8..362805005 100644
--- a/src/Stmt.cc
+++ b/src/Stmt.cc
@@ -2076,7 +2076,10 @@ ValPtr StdFunctionStmt::Exec(Frame* f, StmtFlowType& flow) {
     flow = FLOW_NEXT;
     zeek::Args args = *f->GetFuncArgs();
 
-    func(args);
+    if ( func.index() == 0 )
+        std::get<0>(func)(args);
+    else
+        std::get<1>(func)(args, flow);
 
     return nullptr;
 }
diff --git a/src/Stmt.h b/src/Stmt.h
index 8d9bd1ac7..bdcfb7b88 100644
--- a/src/Stmt.h
+++ b/src/Stmt.h
@@ -782,9 +782,12 @@ protected:
 
 // Statement that calls a std::function. These can be added to a Func body
 // to directly all a C++ method.
+using StdFunction =
+    std::variant<std::function<void(const zeek::Args&)>, std::function<void(const zeek::Args&, StmtFlowType& flow)>>;
+
 class StdFunctionStmt : public Stmt {
 public:
-    StdFunctionStmt(std::function<void(const zeek::Args&)> f) : Stmt(STMT_STD_FUNCTION), func(std::move(f)) {}
+    StdFunctionStmt(StdFunction f) : Stmt(STMT_STD_FUNCTION), func(std::move(f)) {}
 
     ValPtr Exec(Frame* f, StmtFlowType& flow) override;
 
@@ -796,7 +799,7 @@ public:
     TraversalCode Traverse(TraversalCallback* cb) const override { return TC_CONTINUE; }
 
 private:
-    std::function<void(const zeek::Args&)> func;
+    StdFunction func;
 };
```

Reply:
> I could see the function definition just always taking the flow though.

Or that, might be pragmatic enough. We could set FLOW_NEXT before calling func, too.

Reply:
Looks ok with the variant. I pushed a commit with that.